pkg/kv/kvserver/split/weighted_finder.go

```diff
@@ -282,6 +282,22 @@ func (f *WeightedFinder) PopularKeyFrequency() float64 {
 	return popularKeyWeight / totalWeight
 }
 
+// AccessDirection returns a value in [-1, 1] indicating the direction of access
+// requests over time. A negative value implies more traffic is moving to the left
+// (lower keys), a positive value implies more traffic is moving to the right
+// (higher keys), and zero indicates even distribution.
+func (f *WeightedFinder) AccessDirection() float64 {
+	var left, right float64
+	for _, s := range f.samples {
+		left += s.left * s.weight * float64(s.count)
```

Comment:
The sample `left` and `right` weighted counters are bumped by the weight in the `Record` function above (https://github.com/cockroachdb/cockroach/pull/142232/files#diff-640515111864c1a5c78fa051d309cb794af1ed36e4a84a13cb3998380e547c89R119-R123). When determining a split key, the balance score is calculated using only the `left` and `right` weighted counters as well.

I don't think the result of multiplying `s.(left|right) * s.weight * float64(s.count)` is coherent. Consider updating to something like `left += s.left` and likewise for `right`.